src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -38,10 +39,24 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                httpContext.Features.Set(new RequestTelemetry());
+                RequestTelemetry requestTelemetry = new RequestTelemetry();
+                if (string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey))
+                {
+                    this.client.Initialize(requestTelemetry);
```

Comment:
I've changed OnBeginRequest() and EndRequest() to only get their instrumentation key from the client instance. I still call "this.client.Initialize(telemetry)" in EndRequest(), just to ensure that the RequestTelemetry object has the instrumentation key set.